jaggery-modules/rxt/module/scripts/asset/asset.js

```diff
@@ -1650,15 +1682,27 @@ var asset = {};
                     var secondChar = str[1].charAt(0).toLowerCase();
                 } catch (err) {
                     if (typeof secondChar === 'undefined') {
-                        if (str.length > 2){
-                            secondChar =str[0].charAt(1);
+                        if (str.length > 2) {
```

Comment:
Thank you Sameera,
This entire code block has been refactored with new time in lifecycle feature.